libedit/history.c

```diff
@@ -48,7 +48,11 @@ __RCSID("$NetBSD: history.c,v 1.57 2016/04/11 18:56:31 christos Exp $");
 #include <stdarg.h>
 #include <stdlib.h>
 #include <string.h>
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
```

Comment:
The __unix__ macro is also defined on linux, so I would suggest changing the MacOS detection to
#if defined (defined(__APPLE__) && defined(__MACH__))

Currently it breaks the linux build.